nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -345,6 +347,30 @@ def _validate(self) -> None:
     def shard_ids(self) -> List[int]:
         return sorted(self.shard_id_to_manifest.keys())
 
+    def _iter_random_read(self, tar_path, shard_manifest, manifest_path) -> Generator[tuple[dict, bytes], None, None]:
+        with tarfile.open(fileobj=BytesIO(open_best(tar_path, mode="rb").read()), mode="r") as tar:
+            for data in shard_manifest:
+                try:
+                    tar_info = tar.getmember(data)
+                    raw_audio = tar.extractfile(tar_info).read()
+                    yield data, raw_audio, tar_info
+                except KeyError as e:
+                    raise RuntimeError(
```

Comment:
@pzelasko I think you mentioned this is handled by bucketing and not to worry about it ?

Reply:
That wasn't about the mismatches between json and tar files. I meant that you don't need to make everything divisible by 2 or some other number to avoid deadlocks.